src/logics/Logic.h

```diff
@@ -251,6 +251,7 @@ class Logic {
     bool isDisequality(PTRef tr) const;            // { return disequalities.has(term_store[tr].symb()); }
     bool isIte(SymRef tr) const;                   // { return ites.has(tr);          }
     bool isIte(PTRef tr) const;                    // { return ites.has(term_store[tr].symb()); }
+    virtual bool isNonlin(PTRef) const { return false; }
```

Comment:
Does `Logic` really need to know about this?

Reply:
It depends on what and how we want to check.

To check nonlin propperly from the outside I need to have some interface through the logic to interact with ArithLogic, because I cannot call ArithLogic directly from the code of MainSolver for example...
Or can I?

Reply:
I guess we need to do it this way for now.